src/proto/uprotocol/uuid.rs

```diff
@@ -37,140 +41,205 @@ impl std::fmt::Display for UuidConversionError {
     ///
     /// # Returns
     ///
-    /// The number of milliseconds since UNIX EPOCH if this UUID is a uProtocol UUID, `None` otherwise.
+    /// The number of milliseconds since UNIX EPOCH if this UUID is a uProtocol UUID,
+    /// or [`Option::None`] otherwise.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// use uprotocol_sdk::uprotocol::UUID;
+    ///
+    /// // timestamp = 0x018D548EA8E0 (Monday, 29 January 2024, 9:30:52 AM GMT)
+    /// // ver = 0b1000
+    /// let msb = 0x018D548EA8E08000u64;
+    /// // variant = 0b10
+    /// let lsb = 0x8000000000000000u64;
+    /// let creation_time = UUID { msb, lsb, ..Default::default() }.get_time();
+    /// assert_eq!(creation_time.unwrap(), 0x018D548EA8E0_u64);
+    ///
+    /// // timestamp = 1, (invalid) ver = 0b1100
+    /// let msb = 0x000000000001C000u64;
+    /// // variant = 0b10
+    /// let lsb = 0x8000000000000000u64;
+    /// let creation_time = UUID { msb, lsb, ..Default::default() }.get_time();
+    /// assert!(creation_time.is_none());
+    /// ```
     pub fn get_time(&self) -> Option<u64> {
-        let uuid = uuid::Uuid::from(self);
-        uproto_Uuid::try_get_time(&uuid).ok()
+        if self.is_uprotocol_uuid() {
+            // the timstamp is contained in the 48 most significant bits
+            Some(self.msb >> 16)
+        } else {
+            None
+        }
     }
 
     /// Checks if this is a valid uProtocol UUID.
     ///
     /// # Returns
     ///
     /// `true` if this UUID meets the formal requirements defined by the
-    /// [uProtocol spec](https://github.com/eclipse-uprotocol/uprotocol-spec/blob/v1.5.0/basics/uuid.adoc#2-specification).
+    /// [uProtocol spec](https://github.com/eclipse-uprotocol/uprotocol-spec).
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// use uprotocol_sdk::uprotocol::UUID;
+    ///
+    /// // timestamp = 1, ver = 0b1000
+    /// let msb = 0x0000000000018000u64;
+    /// // variant = 0b10
+    /// let lsb = 0x8000000000000000u64;
+    /// assert!(UUID { msb, lsb, ..Default::default() }.is_uprotocol_uuid());
+    ///
+    /// // timestamp = 1, (invalid) ver = 0b1100
+    /// let msb = 0x000000000001C000u64;
+    /// // variant = 0b10
+    /// let lsb = 0x8000000000000000u64;
+    /// assert!(!UUID { msb, lsb, ..Default::default() }.is_uprotocol_uuid());
+    ///
+    /// // timestamp = 1, ver = 0b1000
+    /// let msb = 0x0000000000018000u64;
+    /// // (invalid) variant = 0b01
+    /// let lsb = 0x4000000000000000u64;
+    /// assert!(!UUID { msb, lsb, ..Default::default() }.is_uprotocol_uuid());
+    /// ```
     pub fn is_uprotocol_uuid(&self) -> bool {
-        let uuid = uuid::Uuid::from(self);
-
-        if !matches!(uuid.get_version(), Some(Version::Custom)) {
-            return false;
-        }
-
-        if uuid.get_variant() != Variant::RFC4122 {
-            return false;
-        }
-
-        true
-    }
-}
-
-impl From<uproto_Uuid> for Uuid {
```

Comment:
IIRC this was kinda handy to have since the proto `UUID` was not hashable.

I [used it over here](https://github.com/PLeVasseur/uprotocol-platform-linux-zenoh/blob/65468148fa49389b8cdc56083490f11e8da4b3e0/ustreamer/src/plugins/up_client_full.rs#L372) in some prototype code for the uStreamer.

Is there any way to either keep this or an alternative we can do within `up-rust` to support hashing the `UUID`s?

Reply:
I guess all we need to do is implement the `Hash` trait for `UUID`. I will add that ...